extensions/gc/tasks/GarbageCollectorTask.js

```diff
@@ -188,7 +188,14 @@ class GarbageCollectorTask extends BackbeatTask {
         const { bucket, key, version, oldLocation, newLocation } = entry.getAttribute('target');
 
         async.waterfall([
-            next => this._getMetadata(entry, log, (err, objMD) => {
+            next => this.retry({
+                actionDesc: 'get metadata',
+                logFields: { bucket, key, version },
+                maxRetries: 2,
+                actionFunc: cb => this._getMetadata(entry, log, cb),
+                shouldRetryFunc: err => err.retryable,
+                log,
+            }, (err, objMD) => {
```

Comment:
I think a better approach for this would be having a global retry for the whole function and not a retry per operation. Something like this :
https://github.com/scality/backbeat/blob/e19a1e5a8b9fffc09d536a04fa4176ef2ec13fb4/extensions/replication/tasks/MultipleBackendTask.js#L195-L247

So we rename `_deleteArchivedSourceData` to `_deleteArchivedSourceDataOnce` and `_deleteArchivedSourceData` becomes a new wrapper function that implements the retry on `_deleteArchivedSourceDataOnce`

Reply:
btw, should we also retry `_executeDeleteData` ?

Reply:
Done here : https://github.com/scality/backbeat/pull/2520/commits/8dbcf1115f3227232abe57a90fd3983798cbb46a